### PR TITLE
Fix seeking on contiguous data models

### LINKED ISSUE #2: Cmd+Shift+{arrowkey} causes an 'undefined' index to get sent to data.get()
In the `navigation-model` module, pressing cmd+shift+{arrowkey} starts a seek for the next break in the data in that direction.  Upon reaching an edge, however, checking for the values causes a call to `grid.dataModel.get` with an `undefined` index:

```javascript
while (goForward(newIndex) !== undefined && !isEdgeToSeek(newIndex = goForward(newIndex)));
```

### PR DESCRIPTION
**Pull request by joechrz**
Make sure no 'undefined' values get sent to the data model getter.

Fixes #2

### CODE REVIEW

File: src/modules/navigation-model/navigation-model.spec.js [hunk 1]
```diff
@@ -144,23 +144,21 @@ describe('navigation-model', function() {
                 beforeEachFn.call(this, undefined, undefined, 11, 11);
             });
 
+            function getCoordFromPoint(rowOrCol, coord, amount) {
+                coord[rowOrCol] += amount;
+                return coord;
+            }
+
```
Comment (joechrz) [inline src/modules/navigation-model/navigation-model.spec.js]: Because I had a use for this when I thought the tests were doing something other than they were.  Doesn't hurt to have it tho.

File: src/modules/navigation-model/index.js [hunk 2]
```diff
@@ -73,6 +73,10 @@ module.exports = function(_grid) {
         var isLeftwardEdge, isRightwardEdge, isUpwardEdge, isDownwardEdge, cellHasValue, startedDefined;
         if (isSeek) {
             cellHasValue = function(r, c) {
+                if (r === undefined || c === undefined) {
```
Comment (scamden) [inline src/modules/navigation-model/index.js]: i'd be fine with these being == in case someone passes null (with a comment to document the intention of course)

File: src/modules/navigation-model/navigation-model.spec.js [hunk 3]
```diff
@@ -172,6 +170,19 @@ describe('navigation-model', function() {
                 var firstExpect = getCoordFromCenter(rowOrCol, direction);
                 var secondExpect = getCoordFromCenter(rowOrCol, direction * 4);
                 var finalExpect = getCoordFromCenter(rowOrCol, direction * 5);
+
+                // contiguous data
+                expect(model._navFrom(5, 5, forwardEvent)).toEqual(finalExpect);
```
Comment (scamden) [inline src/modules/navigation-model/navigation-model.spec.js]: nice. this is a sneaky way to do this. i like it

Comment (scamden) [inline src/modules/navigation-model/navigation-model.spec.js]: could we make the comment clearer, like test the "edge cases"

File: src/modules/navigation-model/index.js (hunk 2 above)
Comment (joechrz) [inline src/modules/navigation-model/index.js]: What about just returning false for anything that's not a number?

`isNaN(Number(r))`

File: src/modules/navigation-model/navigation-model.spec.js (hunk 3 above)
Comment (joechrz) [inline src/modules/navigation-model/navigation-model.spec.js]: Something like 
```
// start at the center of the grid and seek; should go all the way to the edge of the grid
```

File: src/modules/navigation-model/index.js (hunk 2 above)
Comment (scamden) [inline src/modules/navigation-model/index.js]: i'd be fine with that with the caveat that this check should be absolutely lightening fast (it gets called a LOT of times) and if creating a number and then checking NaN is as fast as comparing to null i'm good with it. it's probably overly nit picky to check anything other than undefined since that's definitely the result of goForward in the edge case but meh

Comment (joechrz) [inline src/modules/navigation-model/index.js]: Compromise?
```
if ((r === 0 || r > 0) && (c === 0 || c > 0))
```

Comment (scamden) [inline src/modules/navigation-model/index.js]: hmm, could we ever expect negatives.... i guess not for this, but why not
just use >= ?

Comment (joechrz) [inline src/modules/navigation-model/index.js]: Documenting offline convo:

Switching back to the original undefined check for now.  Will add null checks later if necessary.
